internal/mesh/internal/controllers/sidecarproxy/builder/local_app_test.go

```diff
@@ -498,6 +499,54 @@ func TestBuildL4TrafficPermissions(t *testing.T) {
 					},
 				},
 			},
+			"preserves default deny http rules": {
+				defaultAllow: false,
+				workloadPorts: map[string]*pbcatalog.WorkloadPort{
+					"p2": {
+						Protocol: pbcatalog.Protocol_PROTOCOL_HTTP,
+					},
+				},
+				ctp: &pbauth.ComputedTrafficPermissions{
+					AllowPermissions: []*pbauth.Permission{
+						{
+							Sources: []*pbauth.Source{
+								{
+									IdentityName: "foo",
+									Partition:    tenancy.Partition,
+									Namespace:    tenancy.Namespace,
+								},
+							},
+							DestinationRules: []*pbauth.DestinationRule{
+								{
+									PortNames: []string{"p2"},
+									Methods:   []string{"GET"},
+									PathExact: "/bar",
```

Comment:
It would be nice to add tests for:
* All types of path matches
* Header matchers
* L7 without specifying specific ports.

Once we add excludes, we'll need tests for the equivalent cases there too.